src/csm_testing/tests/sat_functional/hwinv/test_hwinv.py

```diff
@@ -0,0 +1,632 @@
+        """Test that `sat hwinv --list-all` prints all expected sections and their headers."""
+        command = 'sat hwinv --list-all'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stderr = proc.stderr.decode()
+
+        section_headers = {
+            "Listing of all nodes": HWINV_NODE_HEADER,
+            "Listing of all chassis": HWINV_CHASSIS_HEADER,
+            "Listing of all HSN boards": HWINV_HSN_BOARDS_HEADER,
+            "Listing of all compute modules": HWINV_COMPUTE_MODULES_HEADER,
+            "Listing of all router modules": HWINV_ROUTER_MODULES_HEADER,
+            "Listing of all node enclosures": HWINV_NODE_ENCLOSURES_HEADER,
+            "Listing of all node enclosure power supplies": HWINV_NODE__ENCLOSURE_POWER_SUPPLIES_HEADER,
+            "Listing of all processors": HWINV_PROCS_HEADER,
+            "Listing of all node accelerators": HWINV_NODE_ACCELS_HEADER,
+            "Listing of all node accelerator risers": HWINV_NODE_ACCEL_RISERS_HEADER,
+            "Listing of all node HSN NICS": HWINV_NODE_HSN_NICS_HEADER,
+            "Listing of all memory modules": HWINV_MEMS_HEADER,
+            "Listing of all drives": HWINV_DRIVES_HEADER,
+            "Listing of all CMM rectifiers": HWINV_CMM_RECTIFIERS_HEADER,
+            "Listing of all node bmcs": HWINV_NODE_BMCS_HEADER,
+            "Listing of all router bmcs": HWINV_ROUTER_BMCS_HEADER,
+            "Listing of all mgmt switches": HWINV_MGMT_SWITCHES_HEADER,
+            "Listing of all cabinet pdus": HWINV_CABINET_PDUS_HEADER,
+            "Listing of all cabinet pdu power connectors": HWINV_CABINET_PDU_POWER_CONNECTORS_HEADER,
+        }
+
+        # Split the output into lines
+        lines = stdout.splitlines()
+        line_count = len(lines)
+
+        # Iterate through lines and process each section
+        for i in range(line_count):
+            line = lines[i].strip()
+
+            if line.startswith("Listing of all ") and line.endswith("################################################################################"):
+                section_title = re.match(r"Listing of all (.+)", line).group(1)
+                self.assertIn(
+                    section_title,
+                    section_headers,
+                    f"Unexpected section title '{section_title}' found in output."
+                )
+
+                # Extract the expected header for this section
+                expected_header = section_headers[section_title]
+
+                # Skip to the header row (3 lines down)
+                header_line_index = i + 3
+                if header_line_index >= line_count:
+                    self.fail(f"Header row for section '{section_title}' not found in output.")
+                header_line = lines[header_line_index].strip()
+
+                # Ensure the header line starts with "|" and contains "xname"
```

Comment:
Also need to account for the possibility of empty tables for components that don't exist on the system like you did in your other tests.